### PR TITLE
muzika: unstable-2023-06-07 -> unstable-2023-11-07

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Requires Gnome-Keyring to be running in the background. For non-Gnome desktops:
```
gnome-keyring-daemon --start -d --components=pkcs11,secrets,ssh
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gtk/4.x.nix [hunk 1]
```diff
@@ -65,7 +65,7 @@ in
 
 stdenv.mkDerivation rec {
   pname = "gtk4";
-  version = "4.10.3";
+  version = "4.11.3";
```
Comment (jtojnar) [inline pkgs/development/libraries/gtk/4.x.nix:68]: This is unstable GTK version.

**Comment by onny** [comment]
@SuperSandro2000 I updated the app and added a now required `libsecret` dependency. Unfortunately the app only starts after executing the Gnome Keyring:

```
gnome-keyring-daemon --start -d --components=pkcs11,secrets,ssh
```

Is this something I could avoid or workaround?

**Comment by SuperSandro2000** [comment]
Note sure, probably by enabling gnome-keyring or some other secrets service like kwallet?

**Comment by onny** [comment]
Somehow I'm getting a strange error while trying to build it

```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'muzika-unstable-2023-06-07'
         whose name attribute is located at /nix/store/9vwf5jm3pszbxz55nv8ail3xp25c42l3-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'muzika-unstable-2023-06-07'

         at /nix/store/9vwf5jm3pszbxz55nv8ail3xp25c42l3-source/pkgs/stdenv/generic/make-derivation.nix:392:7:

          391|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          392|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          393|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: error parsing derivation '/nix/store/87r547pjkig1s6sl2q4k1v3kncx4z2rc-nix-prefetch-git.drv': error: expected string 'D', got ''
```

**Comment by Stunkymonkey** [comment]
maybe @lilyinstarlight can help?

**Comment by lilyinstarlight** [comment]
> `error: error parsing derivation '/nix/store/87r547pjkig1s6sl2q4k1v3kncx4z2rc-nix-prefetch-git.drv': error: expected string 'D', got ''`

Can you check if that .drv file is empty?

You might need to `nix store delete` it or something, but, uh, something might be wrong with your nix store or the backing medium because a malformed .drv file should not happen

**Comment by onny** [comment]
> > `error: error parsing derivation '/nix/store/87r547pjkig1s6sl2q4k1v3kncx4z2rc-nix-prefetch-git.drv': error: expected string 'D', got ''`
> 
> Can you check if that .drv file is empty?
> 
> You might need to `nix store delete` it or something, but, uh, something might be wrong with your nix store or the backing medium because a malformed .drv file should not happen

Oh fixed it, it now builds fine but I still cannot run it. It stuck while starting and one process takes up a lot of CPU power:
```
/nix/store/ljhrkmzkl7324vyckzzsf4d441l170yd-gstreamer-1.22.6/lib/../libexec/gstreamer-1.0/gst-plugin-scanner -l /nix/store/mf2ysx55psrgsi8bpxq3xj5zsn3z74cf-gjs-1.78.0/bin/gjs-console
```

**Comment by onny** [comment]
thanks to a hint of @vixalien, there was one dependency missing, now its working inside a VM

image

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by SuperSandro2000** [comment]
Under wayland the taskbar icon is not shown and falls back to the default wayland icon.

**Comment by SuperSandro2000** [comment]
Also when trying to play music I am always hitting the following error and so far the internet could not help me with that.

```
(com.vixalien.muzika:164332): Gjs-Console-CRITICAL **: 18:08:23.095: error during playback Error from element /GstPlayBin3:playbin3/GstURIDecodeBin3:uridecodebin3/GstURISourceBin:urisourcebin0: Your GStreamer installation is missing a plug-in.
Your GStreamer installation is missing a plug-in.
../gst/playback/gsturisourcebin.c(761): new_demuxer_pad_added_cb (): /GstPlayBin3:playbin3/GstURIDecodeBin3:uridecodebin3/GstURISourceBin:urisourcebin0:
Adaptive demuxer is not streams-aware, check your installation
```

**Comment by onny** [comment]
> Under wayland the taskbar icon is not shown and falls back to the default wayland icon.

Oh I never noticed this since I dont have a task bar :/

> Also when trying to play music I am always hitting the following error and so far the internet could not help me with that.

Hm the gstreamer plugins are in buildInput and working for me. Not sure what causes this issue :(

**Pull request merged**